### PR TITLE
add observers

### PR DESCRIPTION
**Pull request by Kvaz1r**

### CODE REVIEW

File: include/wx/charts/wxchartobservers.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+    IN THE SOFTWARE.
+*/
+
+/*
```
Comment (xleclercq) [inline include/wx/charts/wxchartobservers.h]: Remove the chart.js copyright part as this is a new file we created so it doesn't contain anything from that project.

File: include/wx/charts/wxchartobservers.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+#ifndef _WX_CHARTS_WXCHARTOBSERVERS_H_
+#define _WX_CHARTS_WXCHARTOBSERVERS_H_
+
+#include <wx/vector.h>
+
+template<typename T>
+class wxChartValueObserver
+{
+public:
+    virtual void update(const T &value) = 0;
```
Comment (xleclercq) [inline include/wx/charts/wxchartobservers.h]: Capital u to comply with wxWidgets coding standards.

File: include/wx/charts/wxchartobservers.h [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+};
+
+template<typename T>
+class wxChartObservableValue
+{
+public:
+    wxChartObservableValue() {};
+    virtual ~wxChartObservableValue() {};
+
+    void AddHandler(wxChartValueObserver<T> *observer)
```
Comment (xleclercq) [inline include/wx/charts/wxchartobservers.h]: Call it AddObserver and RemoveObserver to keep it consistent.

File: include/wx/charts/wxchartobservers.h (hunk 2 above)
Comment (xleclercq) [reply include/wx/charts/wxchartobservers.h]: Maybe call it OnUpdate as that is more similar to wxWidgets event handler naming scheme.

**Comment by Kvaz1r** [comment]
fixed

**Pull request merged**